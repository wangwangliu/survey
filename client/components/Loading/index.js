import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import { makeStyles } from '@material-ui/core/styles';

export default function Loading(props) {

  const useStyles = makeStyles({
    root: {
      color: `#546DFF`,
      margin: `0 auto`,
      marginTop: `0.5rem`,
      ...props.style,
    },
  });

  const classes = useStyles();

  return (
    <>
      <CircularProgress size='1.5rem' classes={{ root: classes.root }} />
    </>
  );
}
