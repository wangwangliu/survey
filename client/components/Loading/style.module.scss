.body-mask___bg {
  position: fixed;
  z-index: 2000;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: #fff;
  text-align: center;
}

.loading__facehd {
  position: absolute;
  width: 90px;
  height: 90px;
  line-height: 50px;
  z-index: 2001;
  text-align: center;
  top: 50%;
  left: 50%;
  margin-left: -45px;
  margin-top: -146px;
  background-color: #ededed;
  overflow: hidden;
}

.loading__facehd__wrap {
  position: absolute;
  width: 90px;
  height: 90px;
  line-height: 50px;
  text-align: center;
  top: 0;
  left: 0;
  z-index: 2003;
  background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJIAAACSCAYAAACue5OOAAAZn0lEQVR4Xu2dB9R8V1XFzxZURKoEITRJpEiRjoCIUqMBpDeDlCCIUoJCQDoSOoJIV4lCECKQAAGJIkWagIpKMRZQDE0QRQQEBSyb9fuvM3EYvnn3vjpvJnPWmvX9k7nvvXvvO3PvPefss49sO/ayn4GeM6C9IvWcwf3lh2Zgr0h7RRhkBvaKNMg07m+yV6S9DgwyA3tFGmQa9zfZK9JeBwaZgb0iDTKN+5vsFWmvA4PMwF6RBpnG/U32irTXgUFmYK9Ig0zj/iZnV0UivviJiDgjIj4aER/Pz2cj4t8i4vMR8V8R8bWI+HpEnCMivjMiviMizh8RF4qIwyLiYhFxRERcOiIuHxFXjohznx3V6uyiSCjGH0fEeyLivRHxAUlfGvqF2/62iPj+iLh2RPxwRFw/Iq4SEfz/nZZdVSRWnL+MiNdHxB9GxPsk/d8m3qTtC0fEURFxdETcIiIusIl+jP3MXVOkv46I34mIUyT949iT1/b+ttkafzwi7hwRt4uI72p7j7m23wVF4ixzckScKOlP5jrRq/2yzcp014i4b0T84Lb0e10/t1mR/iUinh8RL5T0r9v8Imyz9T0kIm4GtGcbx7KNioTSPD0iXiDpP7dx0tf12fa1IuKEPE9t1dC2SZFQml/hI+krWzXLLTtrG2vvmRFxnZaXbqz5NigSFtjvRsQvSfrUxmZq4gfbZov76Yh4avqrJu5Bu8fNXZGwvH5e0pvaDavc2jZOxsvmQRdn4vflB3Mdh+P5lpyQ/5vOya+mwxKn5acj4mMRcWZEYC1+SNK/l5/croVtHKBPi4j7zNkfNVdFYhXiIM0qNMg5yPa5IuJHI+JG6SzEaTik+U2fUax35+fNkvCaDyK2fyxdG5cc5IYD32SOivTPEXEvSX/Qd6y2CVfcJiKOiYgbD6w4Nd37SESchgJIIhzTS2yzUp6Ujs1e9xr64rkp0jtx1klCmTpJhilukueL20bEeTvdaPiL3h8RL+W8J4mYXifJs9PxEfGkiPj2TjcZ4aI5KdKvRcRDJf1Pl3Gm1/geEfGwiLhMl3tMdA3jw3h4iqS/7fpM29eLiFdGxCy2ujkoEjGwX5D03C6Tavu7I+Jn06F38S732NA1jPu1rCySWK1ai+3DI+KNGRhuff2QF2xakYBp3EMSv6xWkkv8z0TEkyMCS2tbhUP66yLiwZKwAFtJhlq4HkNiY7JJRfoPApeS3tJ29LavmVbd1jjsKsaIdcqPAocrGKhqSYuU7RLDYiOyKUUizHG0pL9oM2rb+HZw0LGV4QfaRcHSu3/bH1j6xX4jIlilJ5dNKBKAshtJAi9ULbkKsQUCHNt1YbsjRPJISf9dO9jc7l+W7o7aywZpN7UicSa6uaQ/atN728dloBa469lJgMXgDgEWXCW2cQm8IcF0VdcM0WhKRSLMcBdJp9Z2PMMDL9nk3l/b1xHbARO+p6Tfq32G7fNEBD9WvPeTyJSKRMzs12tHlaYt3u2r1l6zw+1wFRwnibBRldgmOQGcOnHE0WUqRQI7dP/a0dgmmArWmgyNvfz/DDxB0mNrJ8Q250kMGgK/o8oUisRAri+J81FREtx1ekR8b7Hx2bPBb0bE/SRxVCiK7TulB7zYtk+DsRXpCxFxzVogfioRe/tc4mN95nbMa7HM7i6piv/T9guA44zZoTEViUHicCT6XRTbl8s9fZu91MVxDtjgWZIeXHM/21i75PNdvaZ9lzZjKtLzJT2gplO2iZFxMCRjdS/1M/BwSYDeipLnTo4Zo6z2YykS6MEr1GSzpomPEpHuvJd2M8CqD3YLF0lRbPPD7hQcL918LEW6Y62/yDZ+pduXOjrR9+TIEfMCQLYtghGDMVMMNyVW612JEB10fGMo0umSblnTS9sPighwSHORK0bE36Uj7yczG5b/N3cB0otR88VSR22z8gNbOWepbZvvh1Yk0oSuJAl2j0axTeQeRCRpzHMQtonLruKsbf9IZsPeISLAfc9VTpV0x5rO2X4egeGatrVthlakJ0p6TOnhtjnw/VVmbZSaT/n9KyT91EEPtH2JiMAZeM85QVxX+gpqAFO/9CNm6/77iLhgqW3t90MqEsvqETUpObbZztjW5iasSteR9L51HUvr54URAS58bvLliLiipE+WOmb7EYl/KjWt+n5IRapy3ycc5E9njCd6iyRy8Eu/ajBRZP6CkZqTvEZS0XjJwC55g4P47YZSpKrVKMFXKBEIxzkLeKm3lzpom4AoDtcfKLWd+HtAg2C5Sz8GEiWq/FClew2lSCdIelzpYbY54HHQm7u8XRKJlEVJP9grIuInio2na/APZBBLIjN4rWTfwTr1XlWHUCQgDhcv5aIlrhgzFd7FbRB8M1AFFiXBZK+aGW7qIZJ+tdR52yAxq0ItTfcaQpE+KelSFR2em8+o1OXXSaoG089QmYguHFlCXdgGqsMK1ovncghF+pykxgObbXLsWY3Iw9oWAaZx+Tb5+7nq4hubDJlYmEzgJliYjWIbko6igTH2isT9ry7pA+seZPsXI6K4zJYGvIHvnyHpoW2ea5utG/fBHLZwSC1wsjZmLw+BWRpiRWKe38xh8yDm2HTkfTAivqfNC5lJW+gFL9Emk4N+2yZZ8W19t4uB5gDcEgStayVXUvgIOh+6h1IkOokZDC77LAKIDIMAwppzLn7pfd1SEojNVmKbFZiVeNPyDkk3LHXCNsoGsVcnGVKR6AAZovADkQAJUA3g/laSay7N5kslQU7RSvJc+JyI+FyScpFZjIUL1Q5nSg65kLnzGZNVhGdy6G6Mf9omSA0veSfpq0gkOeI/wnnH4fSHIuJRfQ9unUYy3kX8KC5SC2tt241UOFYMXiRBV7I/hpZHS4IGZ60klxRMdJ0C030UiSyPW63mqWe2J6nDUNXtilxNEue8USWpeW4NO8vAmKEPSyp6323jDYdQvrV0VSQAYCyXBxJiJUYY3wQR812Q+0oie2MyyQP7MwZ0JVyrBH7rY113VaTXS+KX07RUPit/WZNN/ogPqoLHDP38XN2hLYSlZOH05cxDlB8OBc5dfCDU4NzV5Bh+rKQnFN4ZPN9rkQ9N13ZVpEdJYnBNigSTPb+obRUMB4iwSO7EybgxSfMc7BBK85V15zXbKBJsJHAlrBbPeY8k+Lub3hmoSVLIIC9rJV0V6ShJ+I6aOjVIDKfVaIZpzIGTbex5kggzbJ2kQhH7W+aPgtXkAiWWYNv4v4rugtVJ6aJIgL8uVAKwde3QBt8abGko/4tLk73BPlY/OrmksKaXc9mK8BjbnY4kXRTpY5Iac/IzW4ElEkgtqTK4CXAPsL+zl3MIJ5cNP9OmD+RAfsHkvLIUSqh+izNpaJvEBYD+C1z8IyRBVNa0k9wrIn6r7RC6KBIM940MIbavEBF/k51pdNHnMszeffOEYUDJMoUwwRw+TxvLR3TQIGxfDThsHpbfXVrZ+06E7RcnzpxbQc3M4b1JkfAFAj5sJV0UCYvhsCZ4gm1gqPiSkAdKqgKzJUMt1iAAOEp5jiEo0OPa8A0N0YlEU8KzzYtaCDl0T0lm26o8/rZ9sU1Uf1GC4wxJjbXhkg6nddmyLorEWNb6VRJOiwm52JtbR9B5QE4AWRukAw0hbGG/jCU25QqUYwEJgAKvY1h5siQiAoNLlkIl+IyAmDx3afy2sQ5b7QxdFYlVCVwwqdZnSZ6NIIP6uaX/XQVGXzeDtkkPenYPkDo4KMI4LOubqmt7YoEkFJjH5brQI9donm3GvYh5gmZttEZtU+7iSjX3XrTpqkhcjznJUg0lHVWBOBdRVnOV8eJMSUe26dRq21xuUVC4fmrlMxHxxIh4UVsYSNMDbJ+z7aHcNltFKYYGI9vgefmZQ7hcUfx6pZKtXSzuPooEjxEQEWqoYXkBQaBwzEFyeAnTXaMdtgGZcaZookZmWcYR+syhCgTa5ofAwZwq2WB2iKRziH1aBZQVCCsrTgkFMYr3PKsqLWfEEB9t5KO0fUpEkFlcLV0UiUMhJR+ASHyTJPssyY+rk3YHSa+u7lVDQ9u8TLI2VvdwXtaLIuLxfYrGHDAmUqcglT+ozDrJATcr+Z1sg1SkHlyT3FtSa7O7NKe2Wcnvt9QO9hJ+BGvFNoYSBlO1dFGkkySRtnyg2MZvtIrfeYmkY6t7VWhom5UPCuBFvbXfj4jj+xSJOeiRaThQ1K+J0JNV6eGFF8MW23SY5sx5aUl41QcT2/BN0X8wUAthEeDM2aRIrTOhuygSrBdrydZtXyMJMJc7CsUvmJ5OlY/WvOSjc7vBydYYrun6ZmyTll0qccHLv3CTJZRZrdDJ4ENaFQ7Cx0rivDmYJALjrVDerNz0YZLIEG5SJBy0JE9WSxdFOlfBhwTNHDCT1e0NovbexfyqR5YN86DOCsoWhYGApfmy0nbE5S2IqS5a2k4zZEF18LsvraSsFlTJbA3lLSgCJBFs/zc9oF0NyI3z4KPbzHUXRcIPgaI0aTT0NsvLKW3fIAkU4GSSKwrBy9XEA3LewWI31kuzfbe0TJv6zIpyvtqDfUbyCTF9SdI/DT0ZtkFZcn696Jp7U5YCg6Xp/YHsgGSiWrooElvUwsF14INsA3i7yMqXxNpIjWldSqp6NEsN83zwoQbORPpx5aaVKVOLsNCaSKneJWmjJa5y9QT7zYpXQjgWM3BtY/UCA6qWLop0mVLSoG24dw7KHMGn08oaqB7JSkPbHCjB5TRJsRqBbV7Outw2znw3lETCw0Yk45qPjAgctzUVo4pJk7bxZ1URyS4G3UWRGpMh89eB34Kq0KvCGQVGNxRtVLGNQVCiA365pMYUnPTWA61gYpfTmjlkY7JX0T8PPVjbxCJZNUgrb5NuTT2YxkKLtjn4s61XSxdFqnFoNfkhiHXdrrqHHRvaZltrDFBGxKsk3bnmEUmwBUIBpCJ4dODGyx7jmtv0apNBbfoL+Tqw2C6C36vRErXN4Z+xVksXRaKAcSOE1jbsFoDE1sltJFE+czRZ489afV41T/VoHS3cOHHbBK5ZIVCiztmw+SgY3UpGBuW9WlXn7KJIJ0pqTDWyDVQTyOY6wVqBv4cY3SiS/qw/azg38GxIIlpDJkbp8MpNk9kOCwzlGYrInqjEeUquD9uEvVoVmu6iSEUrJZ1hvKiF5/mgucczzTY5Cg6HB9omiAwWatXqwj1BeYsFTmcK3Wh8BsHghMzcKs89Y1SG+qykdW6BQ/3L3DrcO23OXdFFkXgJFyxF1G2zD5cIO4vZKH3fcP6y4WZadkgS0J3EDdHU/yydAfAMLz1/B2OZXfPct0o6yEl5VnPbWNutjaEuisRDb7CKRVrteCUZO6sR5dob2TL6KtNcrs/S6lAK8iFeCOS2hAoYsvvPlkQW71rpygHQVZFqku1AA7LXlggScAlQcmLUw/eQb6PrvWxTUv0uXa8f4Lq7SQL606RInWiTuypSFVmnbXAvNeUkcOzhkzlpgMma5S0SSUBEYJM8UTXO5JPTudlqHrsqEuGOS0oChdik3UdlSdGaTrHNncBnU5DYmk4uHUjxUUHxR2gClwhnx6HmorYrbdp9RlKRRc420OTWiNauisQAwP80+YoODdI2JngbTkXSo1mdWh/42sxqbduEgKA0QED4cGinMMyidHxVvd4u3uLaPla2a8SR5bsiPsri0Prc1keRivlt2TnM2bbnH7IdUNKnSgL0Nbrk1sMvkQMwikPuHh8KDK8zhekngWjOgk2rEbn0BLJbZWYMPOhjJHFGa+rnbSPiNV2e20eReF6RKiWVqStrKoA4AojPkcS/e4ttsDo4+GCUI2EB3iD+Utl7scrUPgc3wvGlxunPqi5VX7pfh+/xC4HaAM/epEida9/2VaRTJBUzO7JeLbGvti9qMWhWJQKNL48IHKIlllbSwgnYklKD0vDBwcffoUpxgmm6SsXZiNWMGnAo6qaEdPSitWibGCIrcGvpq0iAuojmM1GNYhvMMtjlvgKnAOxpOBT5hbGf40FHeThMojBjV4DEMLiJpKYw0KFx2iZAPUjiQ4+Jqwm0s0J/uOsz+ioSz60i60w4BnSBjZ7VrgOZ+LqiYy+ViNWIEqEHYbWn6jJkqBeriESAaWrkmWzq8BCKxDZz7SbC9kUHbGMVMLGtAoJTzXjlc0g0AH9eTGSwDekVWbablOdKKgH8WDlr8FtrxzGEInFzaspTBKYYgLWN3+Uda/LENjnhNc9m6b+uJLbX0laOhfaRDZfNQNlBOHCeWyu2MThgj2lt9i9uOpQicb/7SKr69SXRJgyqTeiA0rua+nuw2zcuvZSl1RcA/gOn7uTK806WdNdSHwpw4tLlh74fUpGAnl61NjMiMUvAVM9f1dPNNmJluWlNiU+6aZugLOiHVlCMgYfI7gCtM9Zy02oEfIXSpY3wklLfhlQknoVXml8tIZSiJOkUuW69BlF8UL8GFOuhzgq1OoqS+WtYlUOB0YrPXNPgdEnFOGeyvRBf6yVDKxKdeZKk6uS6LHrDQG7QayTjXEwuPkRhjXl8i0enZUqguhXeeZyuH6IdqilH2uuQvej7GIqEb+nWkkBAVkmiAx9P1mllSk3VfXs0wgFK2k4rnNSMCtmwrZHIWipFiitmkHT3MRSJ9welHWcKrLlqsY03Gjf9dasvGrYhLwAPOvnxxZLny4+2fe9kQxm2R93u9lVJRUPGNn49EBq9ZSxFomMcvkFSNmYsrI4gsyZgM8ETPmV5LugKyUKF7KGV2AYqi+GwYI9tdf0IjflBnKPkjrENqLApi7i6a2MqEp0gKs7K1Nr1ntF4MiioedY1h6s0ERgFIBPwVHdi97d9+4jgjDcXJVqM+bwl5IRt/GGDWM1jKxKDwkV/C0ngkjpJpiXD4gHMgZhQZ8dZsqeRYk1tsleX6pg1ddg2feJAPsivutPkrL+ohiGFtLAi2K2mX1MoEv3g8Aouu2hFlDqdmRf4aaAZhgCLzyXX+GywtpgsGG3xp/ABJtwbkpLsdKRyb9JX1DRd4KSI5q8V2xw7iuW3Su+E76dSJJ6Fux522aeU9u6aji+3yXMVIQk+HDJJpf5iKVDZ9jm0T3JPUtIhbZizXEMSlMxNitQWvbr2XlMq0qITgNzIZmikxpnjG7INYhK+JbbXuQssKcQ0mxSpJvewapybUCQ6Bi74QZJgT529pJ+LCDowi06lOjcwyBoM0qkRgbHQWzalSMur0wPmAvQ/aDZt42eBnBM47jZJTZl2DAWK2PSWTSsSA/haOiEpNTGb+mhJYwNpZ2OlzN5vYLwbFGvA2B6spt4cFGkxlbjzoVZ++ibz8pPFBEZXCMtrGNDGU4V+d36MpEZos21qvRCa6i1zUqTFYHASQusLaxikXIRbRpXMyeesAC3hcvWiUZ878s2LGS62wUt9C/F+l37NUZGWxwG4n2g61sXbJMGgP4gk6wYMIFQS4O/cPNN9x/nbkoD6rhXbgN4auQBqOzF3RVodB5BRyjaQtULYhQ/K9eWDfFPpX4IqhlopmOw4L8mUZdXZZtx4zfstUizaBu4yCMf3tinSugkEukLuPSsYWyGrC2Y6SlRiQ6l5KdvY5p2SDiKEPWsstkFZtEJorJuIXVGkbXzRY/e5plpkr1y25QHsFWns17m5+39aUuP2bRuKnUEK6ewVaXMveuwnf01Soxc+ocFfH8LNsVeksV9n/f0xIIjWc6Zb8BT0tSRrMEk11S2Lo9grUnGKRm0AkpEsGsjFvqlEegL7sDYhdYBuZ/GXf/OpYX47ouQyGQpKslekUfVk7c2xMjG7gdR0sprSibpQKlaw5X9TrZLVjFT6Py/4kig7tlrTrfWs7IIiEaujtCa/bMx/UsIhOqdWRx8kZevJrLgAtCh11MjHJ+lyFMmzD6vZ5yvgtq9NXu9efdl2RQLpSG2Nb6loaZtf6TERgfd2k5F7UJrAeuF2euMYYLs+GtClfu1Bz9t2RbpTDaYpA7FknQIJocbG2Bhr8FYA+Ej3oeBhI1NaH0Xoe63tUr3dqkdssyLByQjvT5EBZXkmbFPZkhAJXl22Qfgiu9D+cVueDdLzjPxwHnlvqZ5d1ZuZqFElsX6xN9usSARxYc/vLRmTOzxLqsP8hkVEmg5UhZjjHI45i7FNsZ3ixCOp4ONToBN6D7DhBrbZ/tl2e8k2K9L7JVHRey89ZiDz8oDc9pJtViRwS0dK+kSvGZjZxVmL9ws1ZF5DdN02ZVYpt9pLtlmRGDhYJYoIsvVstdimqDHIzMV2zfYJbAYmfv4ufz5VSx3UNCnp9ISCB/bfXrLtisTg8R8dV0oG7DVLI12c2Sn4vFCgNoSlxMdYiZeVC5Zf/vvMmoKKeS6kgPQgrHK7oEi8ZlYk4LlkRZwmiYPxbCU5G4+NCFK+xyAZo+jiWYqV/+a/AQGihFirlNtqxCu1mcBdUaTlMX8xtzwOkG+qJclqM2ld2iZpPQkFfFh95uZ17zKss67ZRUVanhDMdVhGcA6SdfrBGlrjXjOaFycVNGEa8OA4Qjsx6g/RlynuseuKtDqHwHHhQYIXEici5BIfldQZ3JU1YC+VYRiqJuHgxOFJEHWnVp0mhTy7KdK6uYAtBfpjvNQoFR9WM84TC+AX0XQclDgqcVgelkkFODLnykgyxWJ06Bl7RZpsqnf7QXtF2u33O9no9oo02VTv9oP2irTb73ey0e0VabKp3u0H7RVpt9/vZKPbK9JkU73bD9or0m6/38lGt1ekyaZ6tx+0V6Tdfr+TjW6vSJNN9W4/aK9Iu/1+JxvdNwC5hhHhawXz4gAAAABJRU5ErkJggg==);
  background-size: cover;

}

.loading__facehd__shadow {
  width: 18px;
  height: 88px;
  top: 1px;
  left: 0;
  position: absolute;
  z-index: 2002;
  background: #e0e0e0;
  animation: move 2s infinite;
  -webkit-animation: move 2s infinite;
  transform: skew(-20deg, 0deg);
  /*倾斜 第一个deg参数为x轴，第二个deg参数为y轴**/
  -ms-transform: skew(-20deg, 0deg);
  /*兼容IE9浏览器*/
  -webkit-transform: skew(-20deg, 0deg);
  /*兼容谷歌浏览器*/
  -o-transform: skew(-20deg, 0deg);
  /*兼容opera浏览器*/
  -moz-transform: skew(-20deg, 0deg);
  /*兼容火狐浏览器*/
}

@keyframes move {
  from {
    left: 0px;
  }

  to {
    left: 88px;
  }
}