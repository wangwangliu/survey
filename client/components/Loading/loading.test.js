/* eslint-disable no-undef */
import React from 'react';
import Loading from './index.js';

it('test loading component', () => {
  const wrapper = shallow(<Loading />);
  expect(wrapper).toMatchSnapshot();
});
