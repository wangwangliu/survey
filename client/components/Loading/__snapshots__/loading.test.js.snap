// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test loading component 1`] = `
<div
  className="body-mask___bg"
  id="loading"
>
  <div
    className="loading__facehd"
  >
    <div
      className="loading__facehd__wrap"
    />
    <div
      className="loading__facehd__shadow"
    />
  </div>
</div>
`;
